src/postprocessor/compress.rs

```diff
@@ -92,42 +179,63 @@ impl Postprocessor for Lz4 {
     fn process(&mut self, _ingres_ns: u64, _egress_ns: u64, data: &[u8]) -> Result<Vec<Vec<u8>>> {
         use lz4::EncoderBuilder;
         let buffer = Vec::<u8>::new();
-        let mut encoder = EncoderBuilder::new().level(4).build(buffer)?;
+        let mut encoder = EncoderBuilder::new()
+            .level(self.compression_level)
+            .build(buffer)?;
         encoder.write_all(data)?;
         Ok(vec![encoder.finish().0])
     }
 }
+impl Default for Lz4 {
+    fn default() -> Self {
+        Self {
+            compression_level: 4,
+        }
+    }
+}
 
-#[derive(Clone, Default, Debug)]
-struct Zstd {}
+#[derive(Clone, Debug, Default)]
+struct Zstd {
+    compression_level: i32,
+}
+impl Zstd {
+    pub fn with_config(level: i64) -> Result<Box<dyn Postprocessor>> {
+        if !(-7..=22).contains(&level) {
+            return Err(format!(
+                "Xz2 supports compression level between -7 and 22 but {level} was given"
```

Comment:
This still says Xz2, it should be Zstd here :)
```suggestion
                "Zstd supports compression level between -7 and 22 but {level} was given"
```